docs/source/docker_only.rst

```diff
@@ -0,0 +1,129 @@
+
+Start services needed by dallinger:
+
+.. code-block:: shell
+
+    docker run -d --name dallinger_redis --network=dallinger -v dallinger_redis:/data redis redis-server --appendonly yes
+    docker run -d --name dallinger_postgres --network=dallinger -e POSTGRES_USER=dallinger -e POSTGRES_PASSWORD=dallinger -e POSTGRES_DB=dallinger -v dallinger_postgres:/var/lib/postgresql/data postgres:12
+
+If you have these already created you can remove the existing ones and recreate them. To remove the existing ones run:
+
+.. code-block:: shell
+
+    # Needed if you want to recreate containers
+    docker rm dallinger_redis
+    docker rm dallinger_postgres
+
+or you can start the containers you have:
+
+.. code-block:: shell
+
+    # Needed if you want to start stopped containers
+    docker start dallinger_redis dallinger_postgres
+
+Start adminer:
+
+.. code-block:: shell
+
+    docker run -d -p 8080:8080 --name dallinger_adminer --network dallinger --link dallinger_postgres:db adminer
+
+Connect to adminer by visiting http://localhost:8080
+Select PostgreSQL from the dropdown, and enter `dallinger` as both username and password. You can leave the "database" field blank to get a list of existing databases.
+
+
+Run experiment from docker image
+================================
+
+Enter the experiment directory:
+
+.. code-block:: shell
+
+   cd /path/to/experiment-directory
+
+Create a file named `Dockerfile` with these contents (replace image name in the `FROM` directive to match the dallinger version you depend on in your `constraints.txt` file):
+
+.. code-block::
+
+    # syntax = docker/dockerfile:1.2
+    FROM ghcr.io/dallinger/dallinger:9.1.0a1
+
+    RUN mkdir /experiment
+    COPY requirements.txt /experiment/requirements.txt
+    COPY constraints.txt /experiment/constraints.txt
+
+    WORKDIR /experiment
+
+    RUN python3 -m pip install -e /dallinger[docker]
+
+    # Some experiments might only list dallinger as dependency
+    # If they do the grep command will exit non-0, the pip command will not run
+    # but the whole `RUN` group will succeed thanks to the last `true` invocation
+    RUN grep -v ^dallinger requirements.txt > /tmp/requirements_no_dallinger.txt && \
+        python3 -m pip install -r /tmp/requirements_no_dallinger.txt -c constraints.txt || true
+
+    COPY . /experiment
+    ENV PORT=5000
+
+    CMD dallinger_heroku_web
+
+Build a docker image for the experiment using Buildkit:
+
+.. code-block:: shell
+    EXPERIMENT_IMAGE=my-experiment
+    DOCKER_BUILDKIT=1 docker build . -t ${EXPERIMENT_IMAGE}
+
+Start the development server with docker:
+
+.. code-block:: shell
+
+    docker run --name dallinger --rm -ti -u $(id -u ${USER}):$(id -g ${USER}) -v ${PWD}:/experiment --network dallinger -p 5000:5000 -e FLASK_OPTIONS='-h 0.0.0.0' -e REDIS_URL=redis://dallinger_redis:6379 -e DATABASE_URL=postgresql://dallinger:dallinger@dallinger_postgres/dallinger ${EXPERIMENT_IMAGE} dallinger develop debug
```

Comment:
@silviot do you think we should add `-v $HOME/.dallingerconfig:/root/.dallingerconfig` to this command so that the dallingerconfig file is mapped appropriately?

Reply:
Yes, defintely. I didn't take that into account.